components/scream/src/control/atmosphere_driver.cpp

```diff
@@ -883,6 +908,15 @@ void AtmosphereDriver::initialize_atm_procs ()
 
   const bool restarted_run = m_case_t0 < m_run_t0;
 
+  if (m_surface_coupling_import_data_manager) {
+    m_atm_process_group->setup_surface_coupling_processes(SurfaceCouplingTransferType::Import,
```

Comment:
The `setup_surface_coupling_processes` method seems to just be needed by the SC processes, so injecting it into every atm proc seems a bit hacky. What do you think about the following:

- add it only as a method of the SC classes
- have the AD build the SC import/export atm processes
- the AD passes the SCDataMgr to the import/export atm procs when it creates them
- the AD stores the atm procs as a new atm proc group: [sc_import, atm_proc_group, sc_export]

The latter would require a new ctor for atm proc group, that takes in a set of pre-built atm processes, but should not be too hard.

The overall point being to avoid making base classes more complex only for the sake of a handful of derived ones.

Reply:
I agree. I think the 3rd option (pass SCDataMgr to import/export) would be quite simple. Then in `initialize_impl()` these procs could setup the data. As I look at it now, there is no reason for the separate `setup_surface_coupling_processes()`..